test/whisperkit_test.py

```diff
@@ -0,0 +1,190 @@
+            
+        with futures.ThreadPoolExecutor(max_workers=len(self.devices)) as executor:
+            future_test = { 
+                executor.submit(self.run_test, device):
+                    device for device in self.devices
+            }
+
+            for future in futures.as_completed(future_test):
+                device = future_test[future]
+                try: 
+                    output_json = future.result()
+                except Exception as exc:
+                    print('device %r generated an exception: %s' % (device, exc))
+                else:
+                    if output_json is None:
+                        print('device %r test has failed..')
+                        self.assertTrue(False)
+                    with open(TestWhisperKitAndroid.OUTPUT_FOLDER + device + "_" + TestRunADB.output_file, "w")\
+                        as json_file:
+                        json.dump(output_json, json_file)
+
+
+def download_hg_dataset():
+    test_path = f"{os.path.dirname(os.path.abspath(__file__))}/../test/"
+    prefix = f"{test_path}/{TestWhisperKitAndroid.TEST_DATASET}"
+    if os.path.exists(f"{prefix}/metadata.json"):
+        return prefix
+    
+    hf_hub_download(
+                "argmaxinc/whisperkit-test-data",
+                filename="metadata.json",
+                subfolder=TestWhisperKitAndroid.TEST_DATASET,
+                local_dir=test_path,
+                repo_type="dataset",
+                revision="main",
+    )
+    metadata_file = open(os.path.join(prefix, "metadata.json"))
+    metadata = json.load(metadata_file)
+    metadata_file.close()
+    
+    file_name_filter = ["61-", "121-", "4507-"]
+    for item in metadata:
+        if "audio" not in item:
+            continue
+        if not any([item["audio"].startswith(f) for f in file_name_filter]):
+            continue
+
+        file = item["audio"].split(".")[0] + ".mp3"
+        if os.path.exists(f"{prefix}/{file}"):
+            continue
+        try:
+            hf_hub_download(
+                "argmaxinc/whisperkit-test-data",
+                filename=file,
+                subfolder=TestWhisperKitAndroid.TEST_DATASET,
+                local_dir=test_path,
+                repo_type="dataset",
+                revision="main",
+            )
+            print(f'downloaded {file}')
+        except:
+            print(' ') # this is not an issue, do nothing
+        
+    return prefix
+
+
+class ArgParser(argparse.ArgumentParser):
+    def print_help(self, file = None):
+        super().print_help(file)
+        print("example: ")
+        print("  python3.10 test/whisperkit_test.py -m ./openai_whisper-tiny")
+
+
+if __name__ == "__main__":
+    parser = ArgParser()
+    parser.add_argument("-m", "--model-path", 
+            default="openai_whisper-tiny", type=str,
+            help='path to the models, e.g., openai_whisper-tiny')
+    parser.add_argument("-i", "--input",
```

Comment:
can you add a --report-path argument for the output ? or --output, and check for json suffix (or whatever format the output is in)?

Reply:
please check whisperkit project for conventions and choose as you see fit

Reply:
added -o, --output option